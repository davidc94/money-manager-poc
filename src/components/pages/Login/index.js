import React, { PureComponent } from 'react';
import { View } from 'react-native';
import StyledTextInput from '../../atoms/StyledTextInput';
import StyledText from '../../atoms/StyledText';
import StyledButton from '../../atoms/StyledButton';
import StyledView from '../../atoms/StyledView';

const styles = {
  container: { justifyContent: 'center' },
  centered: { textAlign: 'center' },
};

export default class Login extends PureComponent {
  render() {
    return (
      <StyledView style={styles.container}>
        <View>
          <StyledText weight style={styles.centered}>USERNAME</StyledText>
          <StyledTextInput />
        </View>
        <View>
          <StyledText weight style={styles.centered}>PASSWORD</StyledText>
          <StyledTextInput password />
        </View>
        <StyledButton
          onPress={() => this.props.navigation.navigate('Setup')}
        >
          LOGIN
        </StyledButton>
      </StyledView>
    );
  }
}
