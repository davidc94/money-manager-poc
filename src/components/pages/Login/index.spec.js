import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Login from './index';

configure({ adapter: new Adapter() });

describe('login', () => {
  const render = shallow(<Login />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});