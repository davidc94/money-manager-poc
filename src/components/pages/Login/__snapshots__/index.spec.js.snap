// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component>
          <StyledText
            style={
              Object {
                "textAlign": "center",
              }
            }
            weight={true}
          >
            USERNAME
          </StyledText>
          <StyledTextInput />
        </Component>,
        <Component>
          <StyledText
            style={
              Object {
                "textAlign": "center",
              }
            }
            weight={true}
          >
            PASSWORD
          </StyledText>
          <StyledTextInput
            password={true}
          />
        </Component>,
        <StyledButton
          onPress={[Function]}
        >
          LOGIN
        </StyledButton>,
      ],
      "style": Object {
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StyledText
              style={
                Object {
                  "textAlign": "center",
                }
              }
              weight={true}
            >
              USERNAME
            </StyledText>,
            <StyledTextInput />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "USERNAME",
              "style": Object {
                "textAlign": "center",
              },
              "weight": true,
            },
            "ref": null,
            "rendered": "USERNAME",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StyledText
              style={
                Object {
                  "textAlign": "center",
                }
              }
              weight={true}
            >
              PASSWORD
            </StyledText>,
            <StyledTextInput
              password={true}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "PASSWORD",
              "style": Object {
                "textAlign": "center",
              },
              "weight": true,
            },
            "ref": null,
            "rendered": "PASSWORD",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "password": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "LOGIN",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": "LOGIN",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component>
            <StyledText
              style={
                Object {
                  "textAlign": "center",
                }
              }
              weight={true}
            >
              USERNAME
            </StyledText>
            <StyledTextInput />
          </Component>,
          <Component>
            <StyledText
              style={
                Object {
                  "textAlign": "center",
                }
              }
              weight={true}
            >
              PASSWORD
            </StyledText>
            <StyledTextInput
              password={true}
            />
          </Component>,
          <StyledButton
            onPress={[Function]}
          >
            LOGIN
          </StyledButton>,
        ],
        "style": Object {
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <StyledText
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
                weight={true}
              >
                USERNAME
              </StyledText>,
              <StyledTextInput />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "USERNAME",
                "style": Object {
                  "textAlign": "center",
                },
                "weight": true,
              },
              "ref": null,
              "rendered": "USERNAME",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <StyledText
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
                weight={true}
              >
                PASSWORD
              </StyledText>,
              <StyledTextInput
                password={true}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "PASSWORD",
                "style": Object {
                  "textAlign": "center",
                },
                "weight": true,
              },
              "ref": null,
              "rendered": "PASSWORD",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "password": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "LOGIN",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
